src/commands/maven_restore_artifacts.yml

```diff
@@ -11,13 +11,17 @@ parameters:
     description: 'Path of maven module for the app, or "." for single-app-repo'
     type: string
     default: '.'
+  pom_file:
+    description: 'The pom file to be used for the Maven build'
+    type: string
+    default: 'pom.xml'
 
 steps:
   - when:
       condition: << parameters.prefix >>
       steps:
         - restore_cache:
             keys:
-              - << parameters.prefix >>-{{ checksum "pom.xml" }}-<< parameters.path >>
-              - << parameters.prefix >>-{{ checksum "pom.xml" }}
+              - << parameters.prefix >>-{{ checksum "<< parameters.pom_file >>" }}-<< parameters.path >>
```

Comment:
In order to cover https://github.com/ricardo-ch/ricardo-spring-starters/pull/151#discussion_r952225457